api/loader/fileloader.go

```diff
@@ -293,6 +296,20 @@ func (fl *fileLoader) errIfRepoCycle(newRepoSpec *git.RepoSpec) error {
 // else an error.  Relative paths are taken relative
 // to the root.
 func (fl *fileLoader) Load(path string) ([]byte, error) {
+	if u, err := url.Parse(path); err == nil && strings.HasPrefix(u.Scheme, "http") {
```

Comment:
Would you add a test for this so it doesn't get broken in a regression?

Reply:
It is covered by the example test in https://github.com/kubernetes-sigs/kustomize/blob/master/hack/testExamplesAgainstKustomize.sh

Is it sufficient? @pwittrock

Reply:
Ah, yes I see.  Would you consider adding a unit test as well so this is covered by running `go test` against the directory?

As part of that, we could  have test cases for the following, which would not make sense to put in the examples:

- `http/resource.yaml` gets parsed as a dir
- `https://example.com/resource.yaml` gets parsed as a url
- `httpsnotreal://example.com/resource.yaml` returns an error

These should be handled correctly by the code you provided, but the tests will be helpful for make sure they don't break in the future.

Really appreciate your patience on this.

Reply:
Make sense. Could you help assign the PR to me (not sure if I can help myself to do it).

Reply:
Done